<!DOCTYPE html>
<html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<title>Chapter 3 Forensic parameters | The STRAF Book</title>
<meta name="author" content="Alexandre Gouy and Martin Zieger">
<!-- JS --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.2"></script><script src="https://kit.fontawesome.com/6ecbd6c532.js" crossorigin="anonymous"></script><script src="libs/jquery-3.5.1/jquery-3.5.1.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="libs/bootstrap-4.6.0/bootstrap.min.css" rel="stylesheet">
<script src="libs/bootstrap-4.6.0/bootstrap.bundle.min.js"></script><script src="libs/bs3compat-0.2.4.9003/tabs.js"></script><script src="libs/bs3compat-0.2.4.9003/bs3compat.js"></script><link href="libs/bs4_book-1.0.0/bs4_book.css" rel="stylesheet">
<script src="libs/bs4_book-1.0.0/bs4_book.js"></script><script src="libs/accessible-code-block-0.0.1/empty-anchor.js"></script><script src="https://cdn.jsdelivr.net/autocomplete.js/0/autocomplete.jquery.min.js"></script><script src="https://cdn.jsdelivr.net/npm/mark.js@8.11.1/dist/mark.min.js"></script><!-- CSS --><link rel="stylesheet" href="style.css">
</head>
<body data-spy="scroll" data-target="#toc">

<div class="container-fluid">
<div class="row">
  <header class="col-sm-12 col-lg-3 sidebar sidebar-book"><a class="sr-only sr-only-focusable" href="#content">Skip to main content</a>

    <div class="d-flex align-items-start justify-content-between">
      <h1>
        <a href="index.html" title="">The STRAF Book</a>
      </h1>
      <button class="btn btn-outline-primary d-lg-none ml-2 mt-1" type="button" data-toggle="collapse" data-target="#main-nav" aria-expanded="true" aria-controls="main-nav"><i class="fas fa-bars"></i><span class="sr-only">Show table of contents</span></button>
    </div>

    <div id="main-nav" class="collapse-lg">
      <form role="search">
        <input id="search" class="form-control" type="search" placeholder="Search" aria-label="Search">
</form>

      <nav aria-label="Table of contents"><h2>Table of contents</h2>
        <ul class="book-toc list-unstyled">
<li><a class="" href="index.html">Preface</a></li>
<li><a class="" href="introduction.html"><span class="header-section-number">1</span> Introduction</a></li>
<li><a class="" href="importing-data.html"><span class="header-section-number">2</span> Importing data</a></li>
<li><a class="active" href="forensic-parameters.html"><span class="header-section-number">3</span> Forensic parameters</a></li>
<li><a class="" href="popgen-indices.html"><span class="header-section-number">4</span> Population genetics indices</a></li>
<li><a class="" href="pca-mds.html"><span class="header-section-number">5</span> Multivariate statistics</a></li>
<li><a class="" href="ref-pop.html"><span class="header-section-number">6</span> Reference populations analysis</a></li>
<li><a class="" href="file-conversion.html"><span class="header-section-number">7</span> File conversion</a></li>
<li><a class="" href="snp-analysis.html"><span class="header-section-number">8</span> Analysing SNP data</a></li>
<li><a class="" href="conclusion.html"><span class="header-section-number">9</span> Conclusion</a></li>
<li><a class="" href="references.html"><span class="header-section-number">10</span> References</a></li>
<li><a class="" href="acknowledgments.html"><span class="header-section-number">11</span> Acknowledgments</a></li>
<li class="book-part">Appendix</li>
<li><a class="" href="technical-details.html"><span class="header-section-number">A</span> Technical details</a></li>
<li><a class="" href="strafs-commitments.html"><span class="header-section-number">B</span> STRAF’s commitments</a></li>
</ul>

        <div class="book-extra">
          
        </div>
      </nav>
</div>
  </header><main class="col-sm-12 col-md-9 col-lg-7" id="content"><div id="forensic-parameters" class="section level1">
<h1>
<span class="header-section-number">3</span> Forensic parameters<a class="anchor" aria-label="anchor" href="#forensic-parameters"><i class="fas fa-link"></i></a>
</h1>
<p>In this chapter, we’ll show how to compute forensic parameters using STRAF, and
provide details on how they are computed and should be interpreted.
We’ll introduce a few equations, but please do not be afraid! The goal of
this chapter is to translate each of them into plain English.</p>
<div id="how-to-compute-forensic-parameters-in-straf" class="section level2">
<h2>
<span class="header-section-number">3.1</span> How to compute forensic parameters in STRAF<a class="anchor" aria-label="anchor" href="#how-to-compute-forensic-parameters-in-straf"><i class="fas fa-link"></i></a>
</h2>
<p>Once your data has been uploaded, you can go to the <strong>Forensic parameters</strong> tab
and check the <em>Compute forensics statistics</em> box. The computation will be performed
and a table containing the values per locus will be displayed. The computation is done
per population and overall, a drop-down menu is present to select the population.</p>
<div class="capture">
<div class="inline-figure"><img src="img/capture_forensics_parameters_1.png" width="100%"></div>
</div>
<p>Below the forensic parameters table, You can select the metric you would like
to represent using the drop-down menu.</p>
<div class="capture">
<div class="inline-figure"><img src="img/capture_forensics_parameters_2.png" width="100%"></div>
</div>
</div>
<div id="details-on-the-forensic-parameters" class="section level2">
<h2>
<span class="header-section-number">3.2</span> Details on the forensic parameters<a class="anchor" aria-label="anchor" href="#details-on-the-forensic-parameters"><i class="fas fa-link"></i></a>
</h2>
<div id="random-match-probability-pm" class="section level3">
<h3>
<span class="header-section-number">3.2.1</span> Random match probability (PM)<a class="anchor" aria-label="anchor" href="#random-match-probability-pm"><i class="fas fa-link"></i></a>
</h3>
<p>The <strong>Random match probability</strong>, or probability of matching (PM), is defined as
the probability of observing a random match between two unrelated individuals.</p>
<p><strong>Formula</strong></p>
<p><span class="math display">\[
PM = \sum_i (G_i)^2,
\]</span>
where <span class="math inline">\(G_i\)</span> is the frequency of the genotype <span class="math inline">\(i\)</span> at a given locus in the population.</p>
<p><strong>Interpretation</strong></p>
<p>Computing <span class="math inline">\(PM\)</span> means calculating, for a given locus, the frequency of each
genotypes. Then we take the square of each frequency, i.e. we multiply it by itself.
Finally, we sum the values of each genotype.</p>
<p>The intuition behind it is that if we observe a random match in a population when looking
at a single locus, it means that our two samples have the same genotype at that locus.
In terms of probabilities, sampling a specific genotype in the population has a probability
equal to its frequency. And sampling the same genotype a second time (i.e., observing a match),
is the probability of sampling this genotype multiplied by itself.</p>
<p>As an example, say the genotype “12-14” has a frequency of 5% in the population, the probability of
having a random match between two individuals having the same genotype is 0.05 x 0.05.</p>
<p>To get an overall probability of matching, we sum this over all possible genotypes
in our population.</p>
</div>
</div>
<div id="power-of-discrimination-pd" class="section level2">
<h2>
<span class="header-section-number">3.3</span> Power of Discrimination (PD)<a class="anchor" aria-label="anchor" href="#power-of-discrimination-pd"><i class="fas fa-link"></i></a>
</h2>
<p>The power of discrimination (PD) is defined as the probability of
discriminating between two unrelated individuals.</p>
<p><strong>Formula</strong></p>
<p><span class="math display">\[
PD = 1 - PM
\]</span></p>
<p><strong>Interpretation</strong></p>
<p>PD is simply 1 - PM. Instead of looking at the probability of matching, we are
interested in the probability of “not matching”, i.e. the probability of discrimination.</p>
</div>
<div id="gene-diversity" class="section level2">
<h2>
<span class="header-section-number">3.4</span> Gene diversity<a class="anchor" aria-label="anchor" href="#gene-diversity"><i class="fas fa-link"></i></a>
</h2>
<p><strong>Gene diversity</strong> (<span class="math inline">\(GD\)</span>, sometimes simply <span class="math inline">\(D\)</span>), also called <strong>expected heterozygosity</strong>
(<span class="math inline">\(H_{\mathrm{exp}}\)</span>), is computed using the following estimator:</p>
<p><strong>Formula</strong></p>
<p><span class="math display">\[
  H_{\mathrm{exp}} = GD = \frac{n}{n - 1} \left( 1 - \sum_{i=1}^{n}(p_i)^2 \right),
\]</span></p>
<p>where <span class="math inline">\(n\)</span> is the number of gene copies sampled (i.e., the sample size for haploid samples, and twice the sample size for diploid samples) and <span class="math inline">\(p_i\)</span> is the
frequency of the <span class="math inline">\(i^{th}\)</span> allele in the population.</p>
<p><strong>Interpretation</strong></p>
<p>It is the probability that an individual will be heterozygous at a given locus.</p>
<p>As an example, a value of <span class="math inline">\(GD\)</span> of 0.6 means that there is a 60% chance of being
heterozygote at this locus.</p>
<p>It depends directly on the genetic diversity at this locus, which itself depends on
allele frequencies in your population.</p>
<p>Say we have two alleles in a given population, genetic diversity will be higher
in a population with allele frequencies 0.5 and 0.5 than in a population where
frequencies are 0.1 and 0.9 (as less heterozygotes can be made with rare
alleles). This rationale can be extended to any number of alleles.</p>
</div>
<div id="polymorphism-information-content-pic" class="section level2">
<h2>
<span class="header-section-number">3.5</span> Polymorphism Information Content (PIC)<a class="anchor" aria-label="anchor" href="#polymorphism-information-content-pic"><i class="fas fa-link"></i></a>
</h2>
<p><strong>Formula</strong></p>
<p>The <strong>Polymorphism Information Content</strong> (PIC) is computed as follow:</p>
<p><span class="math display">\[
PIC = 1 - \sum_{i=1}^{n} p_i^2 - \sum_{i=1}^{n-1} \sum_{j=i+1}^{n} 2p_i^2p_j^2,
\]</span>
where <span class="math inline">\(p_i\)</span> and <span class="math inline">\(p_j\)</span> are allele frequencies.</p>
<p><strong>Interpretation</strong></p>
<p>The PIC can be interpreted as:</p>
<ul>
<li><p>the probability that the maternal and paternal alleles of a child are
deducible</p></li>
<li><p>or, the probability of being able to deduce which allele a
parent has transmitted to the child.</p></li>
</ul>
</div>
<div id="power-of-exclusion-pe" class="section level2">
<h2>
<span class="header-section-number">3.6</span> Power of Exclusion (PE)<a class="anchor" aria-label="anchor" href="#power-of-exclusion-pe"><i class="fas fa-link"></i></a>
</h2>
<p><strong>Formula</strong></p>
<p>The <strong>power of exclusion</strong> (<span class="math inline">\(PE\)</span>) is defined as:</p>
<p><span class="math display">\[
PE = h^2\left(1 - 2hH^2\right),
\]</span></p>
<p>where <span class="math inline">\(h\)</span> is the proportion of heterozygous individuals and <span class="math inline">\(H\)</span> the
proportion of homozygous individuals in the population sample.</p>
<p><strong>Interpretation</strong></p>
<p>The power of exclusion depends on the observed proportions of heterozygous and homozygous individuals in a population. These proportions, multiplied as in the equation above, give the <strong>mean exclusion probability for a paternity test</strong>. Intuitively, the more diverse the population is, the greater this value is, as diversity is related to heterozygosity (see above).</p>
</div>
<div id="typical-paternity-index-tpi" class="section level2">
<h2>
<span class="header-section-number">3.7</span> Typical Paternity Index (TPI)<a class="anchor" aria-label="anchor" href="#typical-paternity-index-tpi"><i class="fas fa-link"></i></a>
</h2>
<p>Finally, the typical paternity index (<span class="math inline">\(TPI\)</span>) reflects the “mean PI for
random non-excluded men“ in trio cases (father-mother-child) for a given locus.</p>
<p><strong>Formula</strong></p>
<p>Let <span class="math inline">\(H\)</span> be the frequency of homozygotes, then</p>
<p><span class="math display">\[
TPI = \frac{1}{2H}
\]</span></p>
<p><strong>Interpretation</strong></p>
<p>Unlike the other parameters, the typical paternity index values are not in the 0-1
range and cannot be interpreted as a probability. It is a ratio of 1 divided by
twice the frequency of homozygotes in the population. This is an <strong>odds ratio</strong>,
measuring how many times more likely that a possible father is the actual
father than a randomly selected man in the population, on average (<em>typical</em> case).</p>

</div>
</div>
  <div class="chapter-nav">
<div class="prev"><a href="importing-data.html"><span class="header-section-number">2</span> Importing data</a></div>
<div class="next"><a href="popgen-indices.html"><span class="header-section-number">4</span> Population genetics indices</a></div>
</div></main><div class="col-md-3 col-lg-2 d-none d-md-block sidebar sidebar-chapter">
    <nav id="toc" data-toggle="toc" aria-label="On this page"><h2>On this page</h2>
      <ul class="nav navbar-nav">
<li><a class="nav-link" href="#forensic-parameters"><span class="header-section-number">3</span> Forensic parameters</a></li>
<li><a class="nav-link" href="#how-to-compute-forensic-parameters-in-straf"><span class="header-section-number">3.1</span> How to compute forensic parameters in STRAF</a></li>
<li>
<a class="nav-link" href="#details-on-the-forensic-parameters"><span class="header-section-number">3.2</span> Details on the forensic parameters</a><ul class="nav navbar-nav"><li><a class="nav-link" href="#random-match-probability-pm"><span class="header-section-number">3.2.1</span> Random match probability (PM)</a></li></ul>
</li>
<li><a class="nav-link" href="#power-of-discrimination-pd"><span class="header-section-number">3.3</span> Power of Discrimination (PD)</a></li>
<li><a class="nav-link" href="#gene-diversity"><span class="header-section-number">3.4</span> Gene diversity</a></li>
<li><a class="nav-link" href="#polymorphism-information-content-pic"><span class="header-section-number">3.5</span> Polymorphism Information Content (PIC)</a></li>
<li><a class="nav-link" href="#power-of-exclusion-pe"><span class="header-section-number">3.6</span> Power of Exclusion (PE)</a></li>
<li><a class="nav-link" href="#typical-paternity-index-tpi"><span class="header-section-number">3.7</span> Typical Paternity Index (TPI)</a></li>
</ul>

      <div class="book-extra">
        <ul class="list-unstyled">
          
        </ul>
</div>
    </nav>
</div>

</div>
</div> <!-- .container -->

<footer class="bg-primary text-light mt-5"><div class="container"><div class="row">

  <div class="col-12 col-md-6 mt-3">
    <p>"<strong>The STRAF Book</strong>" was written by Alexandre Gouy and Martin Zieger. It was last built on 2021-10-04.</p>
  </div>

  <div class="col-12 col-md-6 mt-3">
    <p>This book was built by the <a class="text-light" href="https://bookdown.org">bookdown</a> R package.</p>
  </div>

</div></div>
</footer><!-- dynamically load mathjax for compatibility with self-contained --><script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    var src = "true";
    if (src === "" || src === "true") src = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-MML-AM_CHTML";
    if (location.protocol !== "file:")
      if (/^https?:/.test(src))
        src = src.replace(/^https?:/, '');
    script.src = src;
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>
</body>
</html>
