<!DOCTYPE html>
<html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<title>Chapter 1 Importing data | The STRAF Book</title>
<meta name="author" content="Alexandre Gouy and Martin Zieger">
<!-- JS --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.2"></script><script src="https://kit.fontawesome.com/6ecbd6c532.js" crossorigin="anonymous"></script><script src="libs/jquery-3.5.1/jquery-3.5.1.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="libs/bootstrap-4.6.0/bootstrap.min.css" rel="stylesheet">
<script src="libs/bootstrap-4.6.0/bootstrap.bundle.min.js"></script><script src="libs/bs3compat-0.2.4.9003/tabs.js"></script><script src="libs/bs3compat-0.2.4.9003/bs3compat.js"></script><link href="libs/bs4_book-1.0.0/bs4_book.css" rel="stylesheet">
<script src="libs/bs4_book-1.0.0/bs4_book.js"></script><script src="libs/accessible-code-block-0.0.1/empty-anchor.js"></script><script src="https://cdn.jsdelivr.net/autocomplete.js/0/autocomplete.jquery.min.js"></script><script src="https://cdn.jsdelivr.net/npm/mark.js@8.11.1/dist/mark.min.js"></script><!-- CSS --><link rel="stylesheet" href="style.css">
</head>
<body data-spy="scroll" data-target="#toc">

<div class="container-fluid">
<div class="row">
  <header class="col-sm-12 col-lg-3 sidebar sidebar-book"><a class="sr-only sr-only-focusable" href="#content">Skip to main content</a>

    <div class="d-flex align-items-start justify-content-between">
      <h1>
        <a href="index.html" title="">The STRAF Book</a>
      </h1>
      <button class="btn btn-outline-primary d-lg-none ml-2 mt-1" type="button" data-toggle="collapse" data-target="#main-nav" aria-expanded="true" aria-controls="main-nav"><i class="fas fa-bars"></i><span class="sr-only">Show table of contents</span></button>
    </div>

    <div id="main-nav" class="collapse-lg">
      <form role="search">
        <input id="search" class="form-control" type="search" placeholder="Search" aria-label="Search">
</form>

      <nav aria-label="Table of contents"><h2>Table of contents</h2>
        <ul class="book-toc list-unstyled">
<li><a class="" href="index.html">Preface</a></li>
<li><a class="" href="introduction.html">Introduction</a></li>
<li><a class="active" href="importing-data.html"><span class="header-section-number">1</span> Importing data</a></li>
<li><a class="" href="forensic-parameters.html"><span class="header-section-number">2</span> Forensic parameters</a></li>
<li><a class="" href="population-genetics-indices.html"><span class="header-section-number">3</span> Population genetics indices</a></li>
<li><a class="" href="multivariate-statistics.html"><span class="header-section-number">4</span> Multivariate statistics</a></li>
<li><a class="" href="reference-populations-analysis.html"><span class="header-section-number">5</span> Reference populations analysis</a></li>
<li><a class="" href="file-conversion.html"><span class="header-section-number">6</span> File conversion</a></li>
</ul>

        <div class="book-extra">
          
        </div>
      </nav>
</div>
  </header><main class="col-sm-12 col-md-9 col-lg-7" id="content"><div id="importing-data" class="section level1">
<h1>
<span class="header-section-number">1</span> Importing data<a class="anchor" aria-label="anchor" href="#importing-data"><i class="fas fa-link"></i></a>
</h1>
<p>In this chapter, we will explain how to prepare the input file containing
genotypic data, and how to upload it into STRAF.</p>
<div id="str-data-and-straf-input-format" class="section level2">
<h2>
<span class="header-section-number">1.1</span> STR data and STRAF input format<a class="anchor" aria-label="anchor" href="#str-data-and-straf-input-format"><i class="fas fa-link"></i></a>
</h2>
<p>STR data consists in various <strong>observations</strong>: the <strong>genotypes</strong> (number of STR repeats)
of <strong>each individual</strong>, at <strong>each locus</strong>.</p>
<p>In genetics, we potentially observe two values per individual and per locus, if the
markers are diploid, that is two copies are present per sample.</p>
<p><strong>STRAF’s input file</strong> is a <strong>text</strong> file containing the genotypes of each sample:</p>
<ul>
<li><p>The first column, named <strong>ind</strong>, needs to contain the sample ID</p></li>
<li><p>The second column, , named <strong>pop</strong>, contains the population ID (this column must exist even if a single population is studied)</p></li>
<li><p>The next columns correspond to genotypes: for haploid samples, one column per locus must be reported; for diploid data, two columns per locus (with the same name)</p></li>
<li><p>Genotypes must be encoded as numbers (STRAF accepts point alleles)</p></li>
<li><p>Missing data (e.g. null alleles) must be indicated with a “0”.</p></li>
</ul>
<p>For diploid data, the table should look like this:</p>
<div class="inline-table"><table class="table table-sm">
<thead><tr class="header">
<th>ind</th>
<th>pop</th>
<th>Locus1</th>
<th>Locus1</th>
<th>Locus2</th>
<th>Locus2</th>
</tr></thead>
<tbody>
<tr class="odd">
<td>A</td>
<td>Bern</td>
<td>12</td>
<td>14</td>
<td>17</td>
<td>17</td>
</tr>
<tr class="even">
<td>B</td>
<td>Bern</td>
<td>14</td>
<td>14</td>
<td>13</td>
<td>15.2</td>
</tr>
<tr class="odd">
<td>C</td>
<td>Lausanne</td>
<td>12</td>
<td>16</td>
<td>15.2</td>
<td>17</td>
</tr>
</tbody>
</table></div>
<p>For haploid data, the table would look like this:</p>
<div class="inline-table"><table class="table table-sm">
<thead><tr class="header">
<th>ind</th>
<th>pop</th>
<th>Locus1</th>
<th>Locus2</th>
</tr></thead>
<tbody>
<tr class="odd">
<td>A</td>
<td>Bern</td>
<td>12</td>
<td>17</td>
</tr>
<tr class="even">
<td>B</td>
<td>Bern</td>
<td>14</td>
<td>13</td>
</tr>
<tr class="odd">
<td>C</td>
<td>Lausanne</td>
<td>12</td>
<td>15.2</td>
</tr>
</tbody>
</table></div>
</div>
<div id="generating-the-input-data-from-excel" class="section level2">
<h2>
<span class="header-section-number">1.2</span> Generating the input data from Excel<a class="anchor" aria-label="anchor" href="#generating-the-input-data-from-excel"><i class="fas fa-link"></i></a>
</h2>
<p>It only takes a few steps to generate an input file in a format that is suitable
for use in STRAF. From Excel, for example, we can start from a spreadsheet looking
like this:</p>
<center>
<img src="img/capture_excel_1.png" class="capture">
</center>
<p>Then, one simply needs to save this table as a tab-delimited text file. This can be
achieved by clicking on <code>Save As</code> &gt; <code>Text (Tab-delimited) (*.txt)</code></p>
<center>
<img src="img/capture_excel_2.png" class="capture">
</center>
</div>
<div id="uploading-the-data-to-straf" class="section level2">
<h2>
<span class="header-section-number">1.3</span> Uploading the data to STRAF<a class="anchor" aria-label="anchor" href="#uploading-the-data-to-straf"><i class="fas fa-link"></i></a>
</h2>
<p>Once the input file has been prepared, it is possible to upload it into STRAF. Alternatively,
you can <strong>download an example file</strong> by clicking on the link in the sidebar.</p>
<p>Then, you need to select the <strong>ploidy</strong> of your data (for example, <em>Diploid</em> for
autosomal markers, and <em>Haploid</em> for Y-chromosome markers). After that, you can
upload the file by clicking on <em>Browse</em> and selecting the file on your computer.</p>
<p>Once the file is uploaded, a preview will be displayed on the right and all the
analyses will be available. If not, it is likely that an error has occured. If the error
message is not explicit, you can refer to the <strong>Input file checklist</strong> below that
gathers common issues with the input file.</p>
<center>
<img src="img/capture_import_1.png" class="capture">
</center>
</div>
<div id="common-issues" class="section level2">
<h2>
<span class="header-section-number">1.4</span> Common issues<a class="anchor" aria-label="anchor" href="#common-issues"><i class="fas fa-link"></i></a>
</h2>
<p>Even though you’ve been very careful in the generation of STRAF’s input file,
it is possible that you still run into an error after uploading the file to STRAF.
In case STRAF cannot read your input file, we’ve put together a checklist to identify
common issues with the input file.</p>
<div class="interpretation">
<p><strong>Input file checklist</strong></p>
<ul>
<li>Check input parameters in the sidebar: do they actually correspond to the input data?</li>
<li>Check locus names: are they all different for haploid data? Do both columns for a single locus for diploid data have the exact same name?</li>
<li>Check that all missing data have been encoded with a “0”</li>
<li>Try to remove any special characters from sample and locus names</li>
<li>Check for the presence of empty spaces at the end of each line</li>
<li>Check if alleles are exclusively encoded with numbers</li>
<li>Check if values are separated by tabs and not spaces</li>
<li>Check if the first two columns are names “ind” and “pop”</li>
</ul>
</div>
</div>
<div id="having-a-first-look-at-the-data" class="section level2">
<h2>
<span class="header-section-number">1.5</span> Having a first look at the data<a class="anchor" aria-label="anchor" href="#having-a-first-look-at-the-data"><i class="fas fa-link"></i></a>
</h2>
<p>Below the dataset preview, you will be able to generate a plot of allele frequencies
at each locus.</p>
<center>
<img src="img/capture_import_2.png" class="capture">
</center>
<p>You can also generate an allele frequency table, which is standard practice when
reporting new population data in a forensics journal. You can either download
the allele frequencies in a text format (TSV), or as an Excel (XLSX) file. Note
that you have the ability to select a specific population using the drop-down menu
above the table.</p>
<center>
<img src="img/capture_import_3.png" class="capture">
</center>

</div>
</div>
  <div class="chapter-nav">
<div class="prev"><a href="introduction.html">Introduction</a></div>
<div class="next"><a href="forensic-parameters.html"><span class="header-section-number">2</span> Forensic parameters</a></div>
</div></main><div class="col-md-3 col-lg-2 d-none d-md-block sidebar sidebar-chapter">
    <nav id="toc" data-toggle="toc" aria-label="On this page"><h2>On this page</h2>
      <ul class="nav navbar-nav">
<li><a class="nav-link" href="#importing-data"><span class="header-section-number">1</span> Importing data</a></li>
<li><a class="nav-link" href="#str-data-and-straf-input-format"><span class="header-section-number">1.1</span> STR data and STRAF input format</a></li>
<li><a class="nav-link" href="#generating-the-input-data-from-excel"><span class="header-section-number">1.2</span> Generating the input data from Excel</a></li>
<li><a class="nav-link" href="#uploading-the-data-to-straf"><span class="header-section-number">1.3</span> Uploading the data to STRAF</a></li>
<li><a class="nav-link" href="#common-issues"><span class="header-section-number">1.4</span> Common issues</a></li>
<li><a class="nav-link" href="#having-a-first-look-at-the-data"><span class="header-section-number">1.5</span> Having a first look at the data</a></li>
</ul>

      <div class="book-extra">
        <ul class="list-unstyled">
          
        </ul>
</div>
    </nav>
</div>

</div>
</div> <!-- .container -->

<footer class="bg-primary text-light mt-5"><div class="container"><div class="row">

  <div class="col-12 col-md-6 mt-3">
    <p>"<strong>The STRAF Book</strong>" was written by Alexandre Gouy and Martin Zieger. It was last built on 2021-09-26.</p>
  </div>

  <div class="col-12 col-md-6 mt-3">
    <p>This book was built by the <a class="text-light" href="https://bookdown.org">bookdown</a> R package.</p>
  </div>

</div></div>
</footer><!-- dynamically load mathjax for compatibility with self-contained --><script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    var src = "true";
    if (src === "" || src === "true") src = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-MML-AM_CHTML";
    if (location.protocol !== "file:")
      if (/^https?:/.test(src))
        src = src.replace(/^https?:/, '');
    script.src = src;
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>
</body>
</html>
