<!DOCTYPE html>
<html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<title>Chapter 4 Population genetics indices | The STRAF Book</title>
<meta name="author" content="Alexandre Gouy and Martin Zieger">
<!-- JS --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.2"></script><script src="https://kit.fontawesome.com/6ecbd6c532.js" crossorigin="anonymous"></script><script src="libs/jquery-3.5.1/jquery-3.5.1.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="libs/bootstrap-4.6.0/bootstrap.min.css" rel="stylesheet">
<script src="libs/bootstrap-4.6.0/bootstrap.bundle.min.js"></script><script src="libs/bs3compat-0.2.4.9003/tabs.js"></script><script src="libs/bs3compat-0.2.4.9003/bs3compat.js"></script><link href="libs/bs4_book-1.0.0/bs4_book.css" rel="stylesheet">
<script src="libs/bs4_book-1.0.0/bs4_book.js"></script><script src="libs/accessible-code-block-0.0.1/empty-anchor.js"></script><script src="https://cdn.jsdelivr.net/autocomplete.js/0/autocomplete.jquery.min.js"></script><script src="https://cdn.jsdelivr.net/npm/mark.js@8.11.1/dist/mark.min.js"></script><!-- CSS --><link rel="stylesheet" href="style.css">
</head>
<body data-spy="scroll" data-target="#toc">

<div class="container-fluid">
<div class="row">
  <header class="col-sm-12 col-lg-3 sidebar sidebar-book"><a class="sr-only sr-only-focusable" href="#content">Skip to main content</a>

    <div class="d-flex align-items-start justify-content-between">
      <h1>
        <a href="index.html" title="">The STRAF Book</a>
      </h1>
      <button class="btn btn-outline-primary d-lg-none ml-2 mt-1" type="button" data-toggle="collapse" data-target="#main-nav" aria-expanded="true" aria-controls="main-nav"><i class="fas fa-bars"></i><span class="sr-only">Show table of contents</span></button>
    </div>

    <div id="main-nav" class="collapse-lg">
      <form role="search">
        <input id="search" class="form-control" type="search" placeholder="Search" aria-label="Search">
</form>

      <nav aria-label="Table of contents"><h2>Table of contents</h2>
        <ul class="book-toc list-unstyled">
<li><a class="" href="index.html">Preface</a></li>
<li><a class="" href="introduction.html"><span class="header-section-number">1</span> Introduction</a></li>
<li><a class="" href="importing-data.html"><span class="header-section-number">2</span> Importing data</a></li>
<li><a class="" href="forensic-parameters.html"><span class="header-section-number">3</span> Forensic parameters</a></li>
<li><a class="active" href="popgen-indices.html"><span class="header-section-number">4</span> Population genetics indices</a></li>
<li><a class="" href="pca-mds.html"><span class="header-section-number">5</span> Multivariate statistics</a></li>
<li><a class="" href="ref-pop.html"><span class="header-section-number">6</span> Reference populations analysis</a></li>
<li><a class="" href="file-conversion.html"><span class="header-section-number">7</span> File conversion</a></li>
<li><a class="" href="snp-analysis.html"><span class="header-section-number">8</span> Analysing SNP data</a></li>
<li><a class="" href="conclusion.html"><span class="header-section-number">9</span> Conclusion</a></li>
<li><a class="" href="references.html"><span class="header-section-number">10</span> References</a></li>
<li><a class="" href="acknowledgments.html"><span class="header-section-number">11</span> Acknowledgments</a></li>
<li class="book-part">Appendix</li>
<li><a class="" href="technical-details.html"><span class="header-section-number">A</span> Technical details</a></li>
<li><a class="" href="strafs-commitments.html"><span class="header-section-number">B</span> STRAF’s commitments</a></li>
</ul>

        <div class="book-extra">
          
        </div>
      </nav>
</div>
  </header><main class="col-sm-12 col-md-9 col-lg-7" id="content"><div id="popgen-indices" class="section level1">
<h1>
<span class="header-section-number">4</span> Population genetics indices<a class="anchor" aria-label="anchor" href="#popgen-indices"><i class="fas fa-link"></i></a>
</h1>
<p>In this chapter, we will see how to compute some population genetics indices in STRAF.</p>
<div id="computing-population-genetics-parameters-in-straf" class="section level2">
<h2>
<span class="header-section-number">4.1</span> Computing population genetics parameters in STRAF<a class="anchor" aria-label="anchor" href="#computing-population-genetics-parameters-in-straf"><i class="fas fa-link"></i></a>
</h2>
<p>Once you have uploaded your genotypes in STRAF, you can go to the <strong>Population genetics</strong>
tab to compute relevant population genetics indices. It is also possible to perform
a Hardy-Weinberg equilibrium test by checking the relevant box.</p>
<div class="capture">
<div class="inline-figure"><img src="img/capture_popgen_parameters_1.png" width="100%"></div>
</div>
</div>
<div id="details-on-population-genetics-indices" class="section level2">
<h2>
<span class="header-section-number">4.2</span> Details on population genetics indices<a class="anchor" aria-label="anchor" href="#details-on-population-genetics-indices"><i class="fas fa-link"></i></a>
</h2>
<div id="hardy-weinberg-equilibrium" class="section level3">
<h3>
<span class="header-section-number">4.2.1</span> Hardy-Weinberg equilibrium<a class="anchor" aria-label="anchor" href="#hardy-weinberg-equilibrium"><i class="fas fa-link"></i></a>
</h3>
<p>A population is considered at Hardy-Weinberg equilibrium (HWE) when the observed
genotypic frequencies are in agreement with the expectations in an “ideal” population,
which assumes for example random mating in the population.
This is important as the assumptions of the Hardy-Weinberg model
allow to derive quantities such as forensic parameters and population genetic indices.
Therefore, if some assumptions of the model are violated, conclusions drawn from
metrics computed assuming HWE could be challenged.</p>
<p>If a locus presents a significant deviation from HWE, it means that a process
is influencing the distribution of allele and genotype frequencies in the population.
It could for example be due to <strong>inbreeding</strong>, <strong>hidden population structure</strong>,
or <strong>natural selection</strong>.</p>
<p>STRAF reports the <strong>p-value</strong> of a test for HWE. A low p-value indicates a significant
deviation from HWE.</p>
</div>
<div id="heterozygosities" class="section level3">
<h3>
<span class="header-section-number">4.2.2</span> Heterozygosities<a class="anchor" aria-label="anchor" href="#heterozygosities"><i class="fas fa-link"></i></a>
</h3>
<p>In STRAF, several measures of <strong>heterozygosity</strong> are computed. They capture
different aspects of genetic diversity.</p>
<ul>
<li><p>The <strong>expected heterozygosity</strong> (or <strong>Gene diversity</strong>, <span class="math inline">\(H_{exp}\)</span> or <span class="math inline">\(GD\)</span>) has been defined in the previous chapter.</p></li>
<li><p>The <strong>observed heterozygosity</strong> (<span class="math inline">\(H_{obs}\)</span>) is the proportion of heterozygous genotypes at this locus in the population.</p></li>
<li><p>The <strong>total heterozygosity</strong> (<span class="math inline">\(H_T\)</span>) is the heterozygosity expected if all the
individuals in all the subpopulations were behaving as a population at HWE.</p></li>
</ul>
</div>
<div id="f-statistics" class="section level3">
<h3>
<span class="header-section-number">4.2.3</span> F-statistics<a class="anchor" aria-label="anchor" href="#f-statistics"><i class="fas fa-link"></i></a>
</h3>
<p>STRAF reports two <strong>F-statistics</strong>.</p>
<ul>
<li><p>The <span class="math inline">\(F_{\textrm{IS}}\)</span> is a measure of genetic relatedness within a population.
It is sometimes called the <strong>inbreeding coefficient</strong>. High values indicate
a high degree of inbreeding.</p></li>
<li><p>The <span class="math inline">\(F_{\textrm{ST}}\)</span> is a measure of genetic differentiation between populations.
It takes values between 0 (no differentiation) and 1 (full differentiation).</p></li>
</ul>
<div class="interpretation">
<p><strong>One concept, multiple estimators.</strong></p>
<p>Several <strong>estimators</strong> of <span class="math inline">\(F_{\textrm{ST}}\)</span> exist (for example, Weir and Cockerham’s, Nei’s,
Hudson’s <span class="math inline">\(F_{\textrm{ST}}\)</span>). It’s like if each population geneticist had decided
to develop their own estimator!</p>
<p>Why is that? In statistics, what we call an <strong>estimator</strong> is a metric aiming at estimating
a given quantity based on <strong>observed data</strong>. It is important to keep in mind that
these estimators rely on a specific <strong>model</strong>, with underlying assumptions.
It explains why some estimators are more or less reliable depending on the case
and observed data, and each of them has been developed for a different situation.
In the case of <span class="math inline">\(F_{\textrm{ST}}\)</span> for example, different estimators assume
different demographic models.</p>
</div>
</div>
</div>
<div id="linkage-disequilibrium-ld" class="section level2">
<h2>
<span class="header-section-number">4.3</span> Linkage disequilibrium (LD)<a class="anchor" aria-label="anchor" href="#linkage-disequilibrium-ld"><i class="fas fa-link"></i></a>
</h2>
<div id="what-is-linkage-disequilibrium" class="section level3">
<h3>
<span class="header-section-number">4.3.1</span> What is linkage disequilibrium?<a class="anchor" aria-label="anchor" href="#what-is-linkage-disequilibrium"><i class="fas fa-link"></i></a>
</h3>
<p>Linkage disequilibrium is an important quantity to be measured in genetics.
It is defined as the <strong>nonindependence of genotypes at distinct loci</strong>.
It means that it is more likely to observe the co-occurrence of some genotypes
at different loci. This can be influenced by population history.</p>
<p>However, most of the times, LD is explained by the physical proximity between loci. If
two loci are next to each other on the genome, recombination events between them
will be rare and genotypes won’t be shuffled. Genotypes at these loci will be correlated
and linkage disequilibrium will be high. On the other hand, two loci found on two different
chromosomes are not expected to show any LD signals as genotypes will be systematically
shuffled at each generation.</p>
</div>
<div id="how-to-compute-ld-in-straf" class="section level3">
<h3>
<span class="header-section-number">4.3.2</span> How to compute LD in STRAF?<a class="anchor" aria-label="anchor" href="#how-to-compute-ld-in-straf"><i class="fas fa-link"></i></a>
</h3>
<p>It is possible to test for the presence of LD in the dataset you uploaded using
STRAF. After checking the <em>Display pairwise LD p-values matrix</em>, LD tests between
each pair of loci will be performed and p-values will be reported.</p>
<div class="capture">
<div class="inline-figure"><img src="img/capture_popgen_parameters_2.png" width="100%"></div>
</div>
<p><strong>Important note</strong>: Other population genetics software,
<strong>Genepop</strong> and <strong>Arlequin</strong>, implement more reliable
versions of the LD test that should be preferred. They are currently not implemented
in STRAF because of performance limitations. If you need to perform such a test, the
<strong>File conversion</strong> utilities (cf. Chapter 6) should facilitate the workflow.</p>

</div>
</div>
</div>
  <div class="chapter-nav">
<div class="prev"><a href="forensic-parameters.html"><span class="header-section-number">3</span> Forensic parameters</a></div>
<div class="next"><a href="pca-mds.html"><span class="header-section-number">5</span> Multivariate statistics</a></div>
</div></main><div class="col-md-3 col-lg-2 d-none d-md-block sidebar sidebar-chapter">
    <nav id="toc" data-toggle="toc" aria-label="On this page"><h2>On this page</h2>
      <ul class="nav navbar-nav">
<li><a class="nav-link" href="#popgen-indices"><span class="header-section-number">4</span> Population genetics indices</a></li>
<li><a class="nav-link" href="#computing-population-genetics-parameters-in-straf"><span class="header-section-number">4.1</span> Computing population genetics parameters in STRAF</a></li>
<li>
<a class="nav-link" href="#details-on-population-genetics-indices"><span class="header-section-number">4.2</span> Details on population genetics indices</a><ul class="nav navbar-nav">
<li><a class="nav-link" href="#hardy-weinberg-equilibrium"><span class="header-section-number">4.2.1</span> Hardy-Weinberg equilibrium</a></li>
<li><a class="nav-link" href="#heterozygosities"><span class="header-section-number">4.2.2</span> Heterozygosities</a></li>
<li><a class="nav-link" href="#f-statistics"><span class="header-section-number">4.2.3</span> F-statistics</a></li>
</ul>
</li>
<li>
<a class="nav-link" href="#linkage-disequilibrium-ld"><span class="header-section-number">4.3</span> Linkage disequilibrium (LD)</a><ul class="nav navbar-nav">
<li><a class="nav-link" href="#what-is-linkage-disequilibrium"><span class="header-section-number">4.3.1</span> What is linkage disequilibrium?</a></li>
<li><a class="nav-link" href="#how-to-compute-ld-in-straf"><span class="header-section-number">4.3.2</span> How to compute LD in STRAF?</a></li>
</ul>
</li>
</ul>

      <div class="book-extra">
        <ul class="list-unstyled">
          
        </ul>
</div>
    </nav>
</div>

</div>
</div> <!-- .container -->

<footer class="bg-primary text-light mt-5"><div class="container"><div class="row">

  <div class="col-12 col-md-6 mt-3">
    <p>"<strong>The STRAF Book</strong>" was written by Alexandre Gouy and Martin Zieger. It was last built on 2021-10-04.</p>
  </div>

  <div class="col-12 col-md-6 mt-3">
    <p>This book was built by the <a class="text-light" href="https://bookdown.org">bookdown</a> R package.</p>
  </div>

</div></div>
</footer><!-- dynamically load mathjax for compatibility with self-contained --><script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    var src = "true";
    if (src === "" || src === "true") src = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-MML-AM_CHTML";
    if (location.protocol !== "file:")
      if (/^https?:/.test(src))
        src = src.replace(/^https?:/, '');
    script.src = src;
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>
</body>
</html>
